File: sp/src/main/java/wut/sp/Models/Author.java
package wut.sp.Models;

public class Author {
    private String name, surname;

    public Author() {
    }

    public Author(String name) {
        this.name = name;
        this.surname = "";
    }

    public Author(String name, String surname) {
        this.name = name;
        this.surname = surname;
    }

    public void print() {
        System.out.println("Author: " + this.name + " " + this.surname);
    }

    public String getName() {
        return this.name;
    }
}

File: spC++/src/Implement/Models/Author.cpp
#include "Headers/Models/Author.hpp"

Author::Author(std::string name, std::string surname) {
    this->name = name;
    this->surname = surname;
}

Author::Author(std::string name) {
    this->name = name;
}

Author::Author(Author* author) {
    this->name = author->name;
    this->surname = author->surname;
}

void Author::accept(IVisitor* v) {
    v->visitAuthor(this);
}

std::string Author::getName() {
    return this->name + " " + this->surname;
}